cmd/snap-confine/snap-confine.c

```diff
@@ -648,25 +677,31 @@ static void enter_non_classic_execution_environment(sc_invocation *inv,
 	// Init and check rootfs_dir, apply any fallback behaviors.
 	sc_check_rootfs_dir(inv);
 
+	// Set up a device cgroup, unless the snap has been allowed to manage the
+	// device cgroup by itself.
+	if (!is_device_cgroup_self_managed(inv)) {
 	/** Conditionally create, populate and join the device cgroup. */
-	sc_device_cgroup_mode mode = SC_DEVICE_CGROUP_MODE_REQUIRED;
-	const char *non_required_cgroup_bases[] = {
-		/* XXX the list is subject to change if no 'bare' base snaps using
-		 * system-files interface are found in the snap store */
-		"bare",
-		"core", "core16", "core18", "core20", "core22",
-		NULL,
-	};
-	for (const char **non_required_on_base = non_required_cgroup_bases;
-	     *non_required_on_base != NULL; non_required_on_base++) {
-		if (sc_streq(inv->base_snap_name, *non_required_on_base)) {
-			debug("device cgroup not required due to base %s",
-			      *non_required_on_base);
-			mode = SC_DEVICE_CGROUP_MODE_OPTIONAL;
-			break;
+		sc_device_cgroup_mode mode = SC_DEVICE_CGROUP_MODE_REQUIRED;
+		const char *non_required_cgroup_bases[] = {
+			"core", "core16", "core18", "core20", "core22",
```

Comment:
Can you add a comment that explicitly mentions core24 as the cut-off point, so we don't accidentally "add" it here later.

Reply:
this is done